api/src/main/java/io/strimzi/api/kafka/model/KafkaRebalanceSpec.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "goals", "skipHardGoalCheck" })
+@EqualsAndHashCode
+public class KafkaRebalanceSpec implements UnknownPropertyPreserving, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private List<String> goals;
+    private boolean skipHardGoalCheck;
+
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("A list of goals, ordered by decreasing priority, to use for generating and executing the rebalance proposal." +
+            "The supported goals are available at https://github.com/linkedin/cruise-control#goals." +
+            "If an empty goals list is provided, the goals declared in the default.goals Cruise Control configuration parameter are used.")
+    public List<String> getGoals() {
+        return goals;
+    }
+
+    public void setGoals(List<String> goals) {
+        this.goals = goals;
+    }
+
+    @Description("Whether to allow hard goals to be skipped in rebalance proposal generation. Default is false.")
```

Comment:
Out of interest what's the use case for skipping hard goals. On the face of it skipping a hard goal is worse than skipping a soft goal. It might be worth mentioning this in the `@Description`.

Reply:
Of course, you should always hit hard goals but it's anyway a feature that CC allows to you, so why not exposing it?

Reply:
But what's the use case for it? If we don't understand, how can we inform users about when to use this?

Reply:
Imagine by default you care about 4 particular hard goals and this is fine most of the time. However, you notice that your cluster has got into a really strange state (somehow) and that one goal will fix this specific issue. You want to fix only this issue and don't want the other configured `hard.goals` overriding it or preventing a solution being found. In that case you can `skip_hard_goal_check` and just get a proposal for the goal(s) you care about.

Reply:
So why would I not just comment out those hard goals in my CR and do the rebalance? Why this special config for ignoring some other config?

Reply:
I could be wrong, @tomncooper and @kyguy can help on this, but maybe because there are some default hard goals that you don't have in your CR but they are there by default from CC default configuration and in the scenario Tom mentioned you are able to skip checks in this way.

Reply:
Default `hard.goals` are set at the CC server level. If you want to use your own set of goals in a `KafkaRebalance` CR they must include all those configured `hard.goals` or you will get an error. Unless you pass `skip_hard_goals=true`.

If you do not include any goals in a `KafkaRebalance` CR then CC will use the configured `default.goals` (also set at CC server level) to prepare the proposal.

Reply:
OK, I understand now! Thanks! Can we turn that into a pithy sentence or two for the `@Description`? Something like

```java
@Description("Whether to allow the hard goals specified in the Kafka CR to be skipped in rebalance proposal generation. " + 
             "This can be useful when some of those hard goals are preventing a balance solution being found. " + 
             "Default is false.")
```